src/app/reducers/domains/redux/domains.effects.ts

```diff
@@ -1,20 +1,22 @@
 import { Injectable } from '@angular/core';
-import {
-  Actions,
-  Effect
-} from '@ngrx/effects';
+import { Actions, Effect } from '@ngrx/effects';
 import { Observable } from 'rxjs/Observable';
-import * as domainActions from './domains.actions';
-import { Action } from '@ngrx/store';
+import { Account, AccountType } from '../../../shared/models';
+import { AuthService } from '../../../shared/services/auth.service';
+import { State } from '../../index';
+import { Action, Store } from '@ngrx/store';
 import { DomainService } from '../../../shared/services/domain.service';
 import { Domain } from '../../../shared/models/domain.model';
 
+import * as domainActions from './domains.actions';
+import * as fromAuth from '../../auth/redux/auth.reducers';
+
 @Injectable()
 export class DomainsEffects {
-
   @Effect()
   loadDomains$: Observable<Action> = this.actions$
     .ofType(domainActions.LOAD_DOMAINS_REQUEST)
+    .filter((action: domainActions.LoadDomainsRequest) => this.authService.isAdmin())
```

Comment:
very controversial
I think it's better not to dispatch LoadDomainsRequest action
Please discuss with the team